### ISSUE TITLE
Change run-services "Create all service namespaces" to only create namespaces that don't exist.

### ISSUE THREAD

**Issue by joejulian** [samsung-cnct/k2#537]
Currently we build a list of all service namespaces, then delete kube-system from it. This fails with dex as it's installing in the kube-auth namespace which already exists:

```
TASK [roles/kraken.services : Create all service namespaces] *******************
failed: [localhost] (item=kube-auth) => {"changed": true, "cmd": ["/opt/cnct/kubernetes/v1.6/bin/kubectl", "--kubeconfig=/home/jjulian/.kraken/staging/admin.kubeconfig", "create", "namespace", "kube-auth"], "delta": "0:00:00.315612", "end": "2017-06-16 16:29:51.807114", "failed": true, "item": "kube-auth", "rc": 1, "start": "2017-06-16 16:29:51.491502", "stderr": "Error from server (AlreadyExists): namespaces \"kube-auth\" already exists", "stderr_lines": ["Error from server (AlreadyExists): namespaces \"kube-auth\" already exists"], "stdout": "", "stdout_lines": []}
```

Change this task group to build the list of all needed namespaces then remove namespaces that already exist.

**Comment by mikeln** [comment]
Not sure why we have to "pre-build" namespaces?   We should be running `helm install --namespace=<blah>  <chart>` which creates namespaces if missing...(and doesn't error if they are already there).

**Comment by mikeln** [comment]
Ok, just verified.

If you create namespaces via a .yaml file:
```
apiVersion: v1
kind: Namespace
metadata:
  name: mikeln-ns
```
or as part of any other kubernetes .yaml file,

you can use: `kubectl apply -f <that file>.yaml`  which creates if missing and does nothing if they are already there.

**Comment by joejulian** [comment]
I, too, was wondering why this isn't just done by helm.

**Comment by alejandroEsc** [comment]
@mikeln is correct, also if you include that creation of a namespace already defined, k8s spits out an error and then continues to do the remainder of tasks in the yaml. So its a tricky question to answer why helm doesnt do these type of things, nor do i know helm well enough to shout out an opinion, but i would guess that namespacing and placing is a k8s operator management thing.

**Comment by mikeln** [comment]
So we are doing this same song and dance in several places...none of which seem appropriate to me:
```
mikeln-bender:roles Mikel_Nelson$ grep -r "create namespace" *
kraken.fabric/kraken.fabric.flannel/tasks/main.yaml:    {{ kubectl }} --kubeconfig={{ kubefn | expanduser }} create namespace kube-networking
kraken.fabric/kraken.fabric.weave/tasks/main.yaml:    {{ kubectl }} --kubeconfig={{ kubeconfig | expanduser }} create namespace kube-networking
kraken.services/tasks/run-services.yaml:    {{ kubectl }} --kubeconfig={{ kubeconfig | expanduser }} create namespace {{ item }}
```
the namespace create should really be part of the  deployment/daemonset/statefulset/etc  yaml in the first place, and then those should all be created via `apply` instead of `create`.   

Not sure why we are looking to validate a namespace is present before proceeding, this is part of the kubernetes deployment/install already...and we should let kubernetes handle that.

I guess my points: 1) remove all the check for namespace tasks (both before and after), 2) create namespace.yaml for each, or better, put inside the yaml at the top. 3) if the first, then keep the step to create namespace as apply -f namespace.yaml,..if the second, then just make sure to deploy the real yaml via apply -f.

**Comment by mikeln** [comment]
Helm:  so the concept with helm, is that the thing you are deploying with helm can go in whatever namespace you want.   That is why the preferred way to do this with helm is:
`helm install --namepace=<namespace I want> <mychart>`

This is also why, a kubernetes namespace block are not usually included in any yaml for a helm chart.  Though you can, but then you are 'fixing' it to the only namespace it can run in.

**Comment by mikeln** [comment]
looking at our kraken.services role... we are already installing the charts with the --namespace arg.
we don't the all the other steps to pre-create namespaces etc.

**Comment by coffeepac** [comment]
this is resolved by #539
